Source/Other/Extensions/NSAlertExtension.swift

```diff
@@ -35,6 +35,44 @@ import AppKit
 		}
 	}
 
+	/// Creates an alert with primary colored button (also accepts "Enter" key) and cancel button (also accepts escape key), main title and informative subtitle message, and showsSuppressionButton
+		/// - Parameters:
+		///   - title: String for title of the alert
+		///   - message: String for informative message
+		///   - suppression: Bool for showsSuppressionButton
+		///   - defaultKey: String key to set in user defaults
+		///   - primaryButtonTitle: String for main confirm button
+		///   - primaryButtonHandler: Optional block that's invoked when user hits primary button or Enter
+		///   - cancelButtonHandler: Optional block that's invoked when user hits cancel button or Escape
+		/// - Returns: Nothing
+		static func createDefaultAlertWithSuppression(title: String,
+													  message: String,
+													  suppression: Bool,
```

Comment:
Just an idea - can we have `suppressionKey: String?` instead of `suppression` and `defaultKey` - and if they `suppressionKey` exists, then enable suppression, and use it as a key, otherwise disable and don't use it? :) It would be one param less, as they both are connected now, it makes sense to reduce it :)

Reply:
Aw man ... optionals...

Reply:
Like this?

```swift
if let suppressionKey = suppressionKey, !suppressionKey.isEmpty{
    alert.showsSuppressionButton = true
}

// then later...
UserDefaults.standard.set(true, forKey: suppressionKey!)
```

Reply:
yeah

```
static func createDefaultAlertWithSuppression(title: String, message: String, suppressionKey: String? = nil,  primaryButtonTitle: String, primaryButtonHandler: (() -> ())? = nil, cancelButtonHandler: (() -> ())? = nil) {
	let alert = NSAlert()
	alert.messageText = title
	alert.informativeText = message
	if let suppressionKey = suppressionKey {
	 	 alert.showsSuppressionButton = true
	}
	// Order of buttons matters! first button has "firstButtonReturn" return value from runModal()
	 alert.addButton(withTitle: primaryButtonTitle)
	 alert.addButton(withTitle: NSLocalizedString("Cancel", comment: "cancel button"))

	 if alert.runModal() == .alertFirstButtonReturn {
	 	 primaryButtonHandler?()
	} else {
		 cancelButtonHandler?()
	}

	// if they check the box, set the bool
	 if let suppressionButton = alert.suppressionButton, suppressionButton.state == .on {
	 	 UserDefaults.standard.set(true, forKey: defaultKey)
	}
}
```

Reply:
says I must unwrap suppressionKey.... on this line:

`UserDefaults.standard.set(true, forKey: suppressionKey)`

Reply:
ohh then you need the if let again in the second condition

Reply:
then it warns me that the first if let is not needed!

Reply:
```swift
static func createDefaultAlertWithSuppression(title: String,
                                                      message: String,
                                                      suppressionKey: String? = nil,
                                                      primaryButtonTitle: String,
                                                      primaryButtonHandler: (() -> ())? = nil,
                                                      cancelButtonHandler: (() -> ())? = nil) {
            let alert = NSAlert()
            alert.messageText = title
            alert.informativeText = message

            if suppressionKey != nil {
                alert.showsSuppressionButton = true
            }
            // Order of buttons matters! first button has "firstButtonReturn" return value from runModal()
            alert.addButton(withTitle: primaryButtonTitle)
            alert.addButton(withTitle: NSLocalizedString("Cancel", comment: "cancel button"))

            if alert.runModal() == .alertFirstButtonReturn {
                primaryButtonHandler?()
            } else {
                cancelButtonHandler?()
            }

            if let suppressionKey = suppressionKey {
                // if they check the box, set the bool
                if let suppressionButton = alert.suppressionButton,
                   suppressionButton.state == .on {
                    UserDefaults.standard.set(true, forKey: suppressionKey)
                }
            }
        }
```

??

Reply:
yeahm can be like this, and can combine the the ifs at the end :)

Reply:
See what you think now

Reply:
Do not know what's going on with my indentation.

![Screenshot 2020-11-27 at 11 34 18 PM copy](https://user-images.githubusercontent.com/1576190/100465277-faf8ad00-3109-11eb-9157-1e3cbad475f7.png)
![Screenshot 2020-11-27 at 11 34 27 PM copy](https://user-images.githubusercontent.com/1576190/100465286-fe8c3400-3109-11eb-96e6-ba69fc7d1bfe.png)